build/parseSpec.c

```diff
@@ -447,31 +451,32 @@ int readLine(rpmSpec spec, int strip)
 	return PART_ERROR;
     }
 
-    if (!spec->readStack->reading && (lineType->id == LINE_IF)) {
-	match = 0;
-    } else if (lineType->id == LINE_IFARCH) {
+    if (lineType->id & (LINE_IFARCH | LINE_ELIFARCH)) {
 	ARGMATCH(s, "%{_target_cpu}", match);
     } else if (lineType->id == LINE_IFNARCH) {
 	ARGMATCH(s, "%{_target_cpu}", match);
 	match = !match;
-    } else if (lineType->id == LINE_IFOS) {
+    } else if (lineType->id & (LINE_IFOS | LINE_ELIFOS)) {
 	ARGMATCH(s, "%{_target_os}", match);
     } else if (lineType->id == LINE_IFNOS) {
 	ARGMATCH(s, "%{_target_os}", match);
 	match = !match;
-    } else if (lineType->id == LINE_IF) {
-	s += 3;
-        match = parseExpressionBoolean(s);
-	if (match < 0) {
-	    rpmlog(RPMLOG_ERR,
-			_("%s:%d: bad %%if condition\n"),
-			ofi->fileName, ofi->lineNum);
-	    return PART_ERROR;
+    } else if (lineType->id & (LINE_IF | LINE_ELIF)) {
+	s += lineType->textLen;
+	if (((lineType->id == LINE_IF) && (spec->readStack->reading)) ||
+	    ((lineType->id == LINE_ELIF) && (spec->readStack->readable))) {
+	    match = parseExpressionBoolean(s);
```

Comment:
match = .. is at the same indentation level as the continued condition, please fix. Again, helper variable or two can be used to shorten the condition and will likely make it nicer overall.